hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -436,4 +453,28 @@ private void waitForTaskToComplete() throws IOException {
   public synchronized void waitForPendingUploads() throws IOException {
     waitForTaskToComplete();
   }
+
+  /**
+   * Getter method for AbfsOutputStream Statistics.
+   *
+   * @return statistics for AbfsOutputStream.
+   */
+  @VisibleForTesting
+  public AbfsOutputStreamStatisticsImpl getOutputStreamStatistics() {
```

Comment:
why is this being cast rather than returned as is?

Reply:
I think it expects to return AbfsOutputStreamStatisticsImpl, and if we return just AbfsOutputStreamStatistics, it would give rise to incompatible types error.

Reply:
I could cast in the tests where this is called, but I thought it would easier to cast here than casting there as there would be many calls to this method.

Reply:
this would be coding into the implementation/semi-public API something only relevant for testing -and make it very hard to ever change to a different implementation.

Just add some static method in the test suite
```
AbfsOutputStreamStatisticsImp getStreamStatistics(AbfsOutputStream)
```
and you could do the casting in just one place.